frontend/src/metabase/visualizations/visualizations/LinkViz/LinkViz.unit.spec.tsx

```diff
@@ -49,47 +97,138 @@ const setup = (options?: Partial<LinkVizProps>) => {
+    it("should render empty state", () => {
+      setup({
+        isEditing: false,
+        dashcard: emptyLinkDashcard,
+        settings:
+          emptyLinkDashcard.visualization_settings as LinkCardVizSettings,
+      });
+
+      expect(screen.queryByLabelText("link icon")).not.toBeInTheDocument();
+      expect(screen.getByLabelText("question icon")).toBeInTheDocument();
 
-    expect(screen.getByLabelText("link icon")).toBeInTheDocument();
-    expect(screen.getByText("https://example23.com")).toBeInTheDocument();
+      expect(screen.getByText("Choose a link")).toBeInTheDocument();
+    });
   });
 
-  it("should render empty state", () => {
-    setup({
-      isEditing: false,
-      dashcard: emptyLinkDashcard,
-      settings:
-        emptyLinkDashcard.visualization_settings as unknown as LinkCardSettings,
+  describe("entity links", () => {
+    it("shows a link to a pie chart question", () => {
+      setup({
+        isEditing: false,
+        dashcard: questionLinkDashcard,
+        settings:
+          questionLinkDashcard.visualization_settings as LinkCardVizSettings,
+      });
+
+      expect(screen.getByLabelText("pie icon")).toBeInTheDocument();
+      expect(screen.getByText("Question Uno")).toBeInTheDocument();
     });
 
-    expect(screen.queryByLabelText("link icon")).not.toBeInTheDocument();
-    expect(screen.getByLabelText("question icon")).toBeInTheDocument();
+    it("shows a link to a table", () => {
+      setup({
+        isEditing: false,
+        dashcard: tableLinkDashcard,
+        settings:
+          tableLinkDashcard.visualization_settings as LinkCardVizSettings,
+      });
 
-    expect(screen.getByText("Choose a link")).toBeInTheDocument();
+      expect(screen.getByLabelText("database icon")).toBeInTheDocument();
+      expect(screen.getByText("Table Uno")).toBeInTheDocument();
+    });
+
+    it("should be able to search for questions", async () => {
+      const scope = nock(location.origin);
+      setupSearchEndpoints(scope, [
+        {
+          name: "Question Uno",
+          id: 1,
+          display: "pie",
+          model: "card",
+          collection: {},
+        },
+      ] as any);
+
+      setup({
+        isEditing: true,
+        dashcard: searchingDashcard,
+        settings:
+          searchingDashcard.visualization_settings as LinkCardVizSettings,
+      });
+
+      const searchInput = screen.getByPlaceholderText("https://example.com");
+
+      userEvent.click(searchInput);
+
+      expect(await screen.findByLabelText("pie icon")).toBeInTheDocument();
+      expect(await screen.findByText("Question Uno")).toBeInTheDocument();
+    });
+
+    it("clicking a search item should update the entity", async () => {
+      const scope = nock(location.origin);
+      setupSearchEndpoints(scope, [
+        {
+          name: "Question Uno",
+          id: 1,
+          display: "pie",
+          model: "card",
+          collection: {},
+        },
+      ] as any);
```

Comment:
Why do we need type casting here?

Reply:
I haven't actually defined a full `Card` object for this test

Reply:
I think it'd be preferred to use `createMockCard` from `metabase-types/api/mocks` in that case

Reply:
✅  https://github.com/metabase/metabase/pull/28248/commits/e575fc7ed3c02eb186a135a1f36b777bc81d3eec
![Screen Shot 2023-02-17 at 11 11 27 AM](https://user-images.githubusercontent.com/30528226/219744445-5199be4f-e6bc-4158-8a6a-59d4a6de8834.png)